lib/output/writer/dynamodb.go

```diff
@@ -80,6 +84,8 @@ type DynamoDB struct {
 	ttl            time.Duration
 	strColumns     map[string]field.Expression
 	jsonMapColumns map[string]string
+	query          field.Expression
```

Comment:
my primary use case for using an expression is to allow for different statements within the same batch, and unfortunately, I could not figure out another way to support this use case. In my case, I'm looking to replicate change-data-capture workloads to dynamo, and a given batch likely includes inserts + updates + deletes. for example ~code:
```yaml
input:
  # ...

pipeline:
  processors:
    # ...
    - bloblang: |
        root = match meta("operation") {
          "insert" => """INSERT INTO FooTable VALUE { ... }"""
          "update" => """UPDATE FooTable SET ..."""
          "delete" => """DELETE FROM FooTable WHERE ..."""
        }

output:
  aws_dynamodb:
    query: ${!content()}
```

Reply:
Ah I see, the `switch` output and new processor will dispatch message sub-batches for each case, so it'd wouldn't be as efficient as a single batch of mixed queries but it might not be far off when the cardinality of queries is low. I'll try and come up with some alternatives that wouldn't sacrifice efficiency in that way.